osbenchmark/workload_generator/workload_generator.py

```diff
@@ -67,10 +75,20 @@ def extract_mappings_and_corpora(client, output_path, indices_to_extract, number
             logging.getLogger(__name__).exception("Failed to extract index [%s]", index_name)
 
     # That list only contains valid indices (with index patterns already resolved)
+    # For each index, check if docs were requested. If so, extract the number of docs from the map
     for i in indices:
-        docs_to_extract = number_of_docs_requested.get(i["name"], None) if docs_were_requested else None
-        docs_to_extract = int(docs_to_extract) if docs_to_extract is not None else docs_to_extract
-        c = corpus.extract(client, output_path, i["name"], docs_to_extract)
+        custom_docs_to_extract = None
+
+        if docs_were_requested and i["name"] in indices_docs_map:
+            try:
+                custom_docs_to_extract = int(indices_docs_map.get(i["name"]))
+            except ValueError:
+                raise exceptions.InvalidSyntax(
+                    f"Value in <index>:<doc_count> pair is not a string-represented integer: [{indices_docs_map.get(i['name'])}]"
```

Comment:
"not the string representation of an integer" or "the string XYZ cannot be converted to an integer"

Reply:
Will update this. I like this more.